### PR TITLE
Request type colors

### LINKED ISSUE #1064: Use Type colors in Dash reports
# Description

There are colors specified for Types in the 311 Data API. Currently, there is a standard palate used for all Dash reports.

The goal is to use the color for each Type in Dash reports so this is consistent everywhere.

### PR DESCRIPTION
**Pull request by joshuayhwu**
Fixes #1064

  - [x] Up to date with `dev` branch
  - [x] All PR Status checks are successful
  - [x] Peer reviewed and approved
  - [ ] Resolve new issue with duplicate color #1223 
  
Request Type and corresponding hex code available [here](https://github.com/hackforla/311-data/blob/b0717ad25eab4ef25a2f26da187d9d8905ab5f77/server/api/alembic/seeds/types.csv) are consistent according to the table here: 

Any questions? See the [getting started guide](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md)

### CODE REVIEW

File: server/dash/dashboards/neighborhood_recent.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ def populate_options():
     y="srnumber",
     color="typeName",
     color_discrete_sequence=DISCRETE_COLORS,
+    color_discrete_map = DISCRETE_COLORS_MAP,
```
Comment (nichhk) [inline server/dash/dashboards/neighborhood_recent.py:69]: nit: no spaces around "=" when using named arguments

File: server/dash/dashboards/neighborhood_recent.py [hunk 2]
```diff
@@ -83,7 +84,9 @@ def populate_options():
     figure_df,
     names="typeName",
     values="srnumber",
+    color = 'typeName',
     color_discrete_sequence=DISCRETE_COLORS,
+    color_discrete_map = DISCRETE_COLORS_MAP,
```
Comment (nichhk) [inline server/dash/dashboards/neighborhood_recent.py:89]: ditto

File: server/dash/dashboards/overview.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import textwrap
 
+
```
Comment (nichhk) [inline server/dash/dashboards/overview.py:3]: delete extra new line

File: server/dash/design.py [hunk 4]
```diff
@@ -47,6 +47,24 @@
 }
 
 
+# TODO: Fixing multiple streetlights and waterwaste color
+# Need to add another discrete color
+DISCRETE_COLORS_MAP = {
```
Comment (nichhk) [inline server/dash/design.py:52]: Just double checking, did you get these from https://github.com/hackforla/311-data/blob/b0717ad25eab4ef25a2f26da187d9d8905ab5f77/server/api/alembic/seeds/types.csv? If so, could you leave a comment to that effect?

File: server/dash/dashboards/recent.py [hunk 5]
```diff
@@ -8,7 +8,7 @@
 import plotly.graph_objects as go
 
 from config import API_HOST
-from design import CONFIG_OPTIONS, DISCRETE_COLORS, LABELS, apply_figure_style
+from design import CONFIG_OPTIONS, DISCRETE_COLORS, LABELS, apply_figure_style, DISCRETE_COLORS_MAP
```
Comment (nichhk) [inline server/dash/dashboards/recent.py:11]: same note about alphabetized ordering

File: server/dash/dashboards/recent.py [hunk 6]
```diff
@@ -33,6 +33,7 @@
     y="counts",
     color="type_name",
     color_discrete_sequence=DISCRETE_COLORS,
+    color_discrete_map = DISCRETE_COLORS_MAP,
```
Comment (nichhk) [inline server/dash/dashboards/recent.py:36]: spacing

File: server/dash/dashboards/recent.py [hunk 7]
```diff
@@ -50,7 +51,9 @@
     pie_df,
     names="type_name",
     values="counts",
+    color='type_name',
     color_discrete_sequence=DISCRETE_COLORS,
+    color_discrete_map = DISCRETE_COLORS_MAP,
```
Comment (nichhk) [inline server/dash/dashboards/recent.py:56]: spacing

File: server/dash/design.py [hunk 8]
```diff
@@ -47,6 +47,24 @@
 }
 
 
+# TODO: Fixing multiple streetlights and waterwaste color
```
Comment (nichhk) [inline server/dash/design.py:50]: Can you open an issue for this, and reference it here?

File: server/dash/dashboards/overview.py [hunk 9]
```diff
@@ -1,7 +1,9 @@
 import textwrap
 
+ 
 import dash_core_components as dcc
 import dash_html_components as html
+# from dash import Dash, html, dcc
```
Comment (nichhk) [inline server/dash/dashboards/overview.py:6]: Delete? It also looks like there are some other test-related lines to delete.

**Review by nichhk — CHANGES_REQUESTED** [review]
Thanks for working on this Josh!

Most of my comments are style nits. I think a linter would automatically fix most of them. Speaking of which, we should decide on a Python linter to use, and make sure that everyone installs it. We can also try to make it part of the "Checks" process.

File: server/dash/dashboards/neighborhood_recent.py (hunk 1 above)
Comment (joshuayhwu) [reply server/dash/dashboards/neighborhood_recent.py:69]: Resolve in newest commit

File: server/dash/design.py [hunk 10]
```diff
@@ -57,7 +75,7 @@ def apply_figure_style(fig):
         title_yanchor='top',
         font_family="Roboto, Arial",
         font_color="#ECECEC",
-        colorway=DISCRETE_COLORS,
+        #colorway=DISCRETE_COLORS,
```
[the remark this answers is not in the document]
Comment (joshuayhwu) [reply server/dash/design.py]: Thanks for the review - resolve with newest commit

File: server/dash/design.py (hunk 4 above)
Comment (joshuayhwu) [reply server/dash/design.py:52]: Added link to PR description

File: server/dash/design.py (hunk 8 above)
Comment (joshuayhwu) [reply server/dash/design.py:50]: Resolve with #1223

**Comment by joshuayhwu** [comment]
> Thanks for working on this Josh!
> 
> Most of my comments are style nits. I think a linter would automatically fix most of them. Speaking of which, we should decide on a Python linter to use, and make sure that everyone installs it. We can also try to make it part of the "Checks" process.
> 
> EDIT: I see that we're already [using](https://github.com/hackforla/311-data/blob/b0717ad25eab4ef25a2f26da187d9d8905ab5f77/.github/workflows/build_test_backend.yml#L28) flake8 for some of the codebase, but maybe just server/api? I'll open an issue for this.

Thnks for the comprehensive feedback! I updated the code to make sure they stay consistent. I will work on the duplicate color file later. Apologies for the style inconsistencies!

File: server/dash/design.py (hunk 4 above)
Comment (nichhk) [reply server/dash/design.py:52]: Oops sorry, I meant to leave a comment in the code. That way, when someone reads the code, they can cross-reference it with types.csv.

File: server/dash/dashboards/overview.py [hunk 11]
```diff
@@ -88,13 +89,18 @@
 print(" * Downloading data for dataframe")
 query_string = "/reports?field=type_name&filter=created_date>=2016-01-01"
 df3 = pd.read_json(API_HOST + query_string)
-df3 = df3.groupby(['type_name'])['counts'].sum().to_frame()
-df3.index = df3.index.map(lambda x: '<br>'.join(textwrap.wrap(x, width=16)))
+df3 = df3.groupby(['type_name'], as_index=False)['counts'].sum()
+#df3 = df3.groupby(['type_name'])['counts'].sum().to_frame()
+#df3.index = df3.index.map(lambda x: '<br>'.join(textwrap.wrap(x, width=16)))
+print(df3['type_name'])
+print(" * TESTING IF CHANGES ARE ACTUALLY DEPLOYED ON DOCKER")
```
Comment (nichhk) [inline server/dash/dashboards/overview.py:96]: Do we still need these lines?

**Review by nichhk — COMMENTED** [review]
Just a couple more requests. I'll approve the changes now so that you don't need to wait on me after fixing the changes. Thanks!

**Review by nichhk — APPROVED** [review]

**Comment by joshuayhwu** [comment]
> Just a couple more requests. I'll approve the changes now so that you don't need to wait on me after fixing the changes. Thanks!

Thanks so much - apologies for missing that!

**Pull request merged**